core/src/main/java/me/mykindos/betterpvp/core/redis/Redis.java

```diff
@@ -1,19 +1,54 @@
 package me.mykindos.betterpvp.core.redis;
 
+import com.google.common.base.Preconditions;
 import com.google.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
 import org.bukkit.configuration.ConfigurationSection;
 import org.bukkit.configuration.file.FileConfiguration;
+import org.jetbrains.annotations.NotNull;
 import redis.clients.jedis.JedisPool;
 import redis.clients.jedis.JedisPoolConfig;
 
 import java.time.Duration;
 
 @Singleton
+@Slf4j
 public class Redis {
 
     private JedisPool pool;
 
     public void credentials(final RedisCredentials redisCredentials) {
+        Preconditions.checkNotNull(redisCredentials, "Redis credentials cannot be null!");
+        Preconditions.checkState(pool == null, "Redis credentials have already been set!");
+
+        final JedisPoolConfig poolConfig = getJedisPoolConfig();
+        if (redisCredentials.getPassword() == null || redisCredentials.getPassword().isEmpty()) {
+            // Passwordless redis
+            this.pool = new JedisPool(
+                    poolConfig,
+                    redisCredentials.getHost(),
+                    redisCredentials.getPort(),
+                    0,
+                    null,
```

Comment:
i dont think this will actually make any difference because config.getString("password") was nullable, you can even see that you are checking if redisCredentials.getPassword() == null, so passing that in would be same as passing null

Reply:
good catch